examples/custom-types/src/lib.rs

```diff
@@ -40,7 +46,7 @@ pub struct CustomTypesDemo {
 
 pub fn get_custom_types_demo(v: Option<CustomTypesDemo>) -> CustomTypesDemo {
     v.unwrap_or_else(|| CustomTypesDemo {
-        url: Url::parse("http://example.com/").unwrap(),
+        url: url::Url::parse("http://example.com/").unwrap().into(),
```

Comment:
I'm not that keen on this change - it forces us back into using the wrapper explicitly, which the `UniffiCustomTypeConverter` was added explicitly to avoid.

The other direction of this can be demonstrated by adding to the udl, say:
`string rt_url(Url url);`
and the matching impl:
`pub fn rt_url(url: url::Url) -> String { url.into() }`

works on main and fails with this patch. For our use-case, this would mean that every public function we have, which uses `url::Url` can't be exposed directly and will need to change to use the wrapper type or jump through other hoops.

IOW, we previously had a solution that involved wrapper types, but added `UniffiCustomTypeConverter` because the wrapper types are a PITA. I'm 100% up for changing the implementation, but not really up for regressing the capability.

Reply:
eg, [this udl](https://github.com/mozilla/application-services/blob/7155bbb4a9b091a272cae01f98e8ceefe9235160/components/places/src/places.udl#L65) has [this impl](
https://github.com/mozilla/application-services/blob/7155bbb4a9b091a272cae01f98e8ceefe9235160/components/places/src/ffi.rs#L271) - and obviously every other use of `Url` in that udl.